### PR TITLE
Step registry caches duplicate step file locations

### PR DESCRIPTION
**Pull request by nehashri**

### CODE REVIEW

File: src/step-registry.js [hunk 1]
```diff
@@ -42,11 +47,13 @@ StepRegistry.prototype.getStepTexts = function () {
   });
 };
 
-StepRegistry.prototype.getStepPositions = function(filePath) {
+StepRegistry.prototype.getStepPositions = function (filePath) {
   var stepPositions = [];
   for (var step in this.registry) {
-    if (this.registry[step].filePath === filePath){
-      stepPositions.push({stepValue : step, lineNumber: this.registry[step].line});
+    for (var i = 0; i < this.registry[step].fileLocations.length; i++) {
```
Comment (kashishm) [inline src/step-registry.js]: Any specific reason for not using `forEach`?

File: src/step-registry.js [hunk 2]
```diff
@@ -11,15 +11,20 @@ var StepRegistry = function () {
...
     generalisedText: generalisedText,
-    filePath: filePath,
-    line: line,
+    fileLocations: [
+      {
+        filePath: filePath,
+        line: line
+      }
+    ],
     count: 1,
```
Comment (kashishm) [inline src/step-registry.js]: Why is this required? It could be a method which returns `fileLocations.length`.

File: src/step-registry.js (hunk 1 above)
Comment (nehashri) [reply src/step-registry.js]: I tried foreach first. The jshint linter we are using kept on telling me that `using outside variables inside function scope will lead to confusing syntax`. I tried a few things. This looked cleaner to me.
